### PR TITLE
add P3018 error and fix meta

### PR DESCRIPTION
**Pull request by Jolg42**

### CODE REVIEW

**Comment by Jolg42** [comment]
Fixes
- put common up
- remove "client" mention since error codes are about many engines
- group Prisma Client error reference 
- add P3018 error

**Comment by Jolg42** [comment]
Seems that I broke the layout 🔥 
See https://deploy-preview-1571--prisma2-docs.netlify.app/docs/reference/api-reference/error-reference#prisma-migrate---migration-engine

File: content/400-reference/200-api-reference/075-error-reference.mdx [hunk 1]
```diff
@@ -72,80 +138,17 @@ Prisma Client throws a `PrismaClientKnownRequestError` exception if the query en
...
 | `P3017`   | "The migration {migration_name} could not be found. Please make sure that the migration exists, and that you included the whole name of the directory. (example: \"20201207184859_initial_migration\")"                                                                                                                                                                 |
+| `P3018`   | "A migration failed to apply. New migrations can not be applied before the error is recovered from. Read more about how to resolve migration issues in a production database: https://pris.ly/d/migrate-resolve <br />Migration name: {migration_name}<br />Database error code: {database_error_code}<br />Database error: {database_error}"                                                                                                                                                                 |
```
Comment (Jolg42) [inline content/400-reference/200-api-reference/075-error-reference.mdx]:
```suggestion
| `P3014`   | "The datasource provider `{provider}` specified in your schema does not match the one specified in the migration_lock.toml. You will encounter errors when you try to apply migrations generated for a different provider. Please archive your current migration directory at a different location and start a new migration history with `prisma migrate dev`. "<br /> |
|
| `P3015`   | "Could not find the migration file at `{migration_file_path}`. Please delete the directory or restore the migration file."<br />                                                                                                                                                                                                                                        |
| `P3016`   | "The fallback method for database resets failed, meaning Migrate could not clean up the database entirely. Original error: {error_code}\n{inner_error}"<br />                                         |
| `P3017`   | "The migration {migration_name} could not be found. Please make sure that the migration exists, and that you included the whole name of the directory. (example: \"20201207184859_initial_migration\")"<br />                                                                                                                             |
| `P3018`   | "A migration failed to apply. New migrations can not be applied before the error is recovered from. Read more about how to resolve migration issues in a production database: https://pris.ly/d/migrate-resolve <br />Migration name: {migration_name}<br />Database error code: {database_error_code}<br />Database error: {database_error}"<br /> |                                                                                                                                      |
```

File: content/400-reference/200-api-reference/075-error-reference.mdx [hunk 2]
```diff
@@ -73,79 +139,16 @@ Prisma Client throws a `PrismaClientKnownRequestError` exception if the query en
 | `P3012`   | "Migration `{migration_name}` cannot be rolled back because it is not in a failed state. "<br />                                                                                                                                                                                                                                                                        |
 | `P3013`   | "Datasource provider arrays are no longer supported in migrate. Please change your datasource to use a single provider. Read more at https://pris.ly/multi-provider-deprecation "<br />                                                                                                                                                                                 |
 | `P3014`   | "The datasource provider `{provider}` specified in your schema does not match the one specified in the migration_lock.toml. You will encounter errors when you try to apply migrations generated for a different provider. Please archive your current migration directory at a different location and start a new migration history with `prisma migrate dev`. "<br /> |
+|
```
Comment (Jolg42) [inline content/400-reference/200-api-reference/075-error-reference.mdx]:
```suggestion
```

File: content/400-reference/200-api-reference/075-error-reference.mdx [hunk 3]
```diff
@@ -74,78 +140,14 @@ Prisma Client throws a `PrismaClientKnownRequestError` exception if the query en
...
+| `P3016`   | "The fallback method for database resets failed, meaning Migrate could not clean up the database entirely. Original error: {error_code}\n{inner_error}"<br />                                         |
+| `P3017`   | "The migration {migration_name} could not be found. Please make sure that the migration exists, and that you included the whole name of the directory. (example: \"20201207184859_initial_migration\")"<br />                                                                                                                             |
+| `P3018`   | "A migration failed to apply. New migrations can not be applied before the error is recovered from. Read more about how to resolve migration issues in a production database: https://pris.ly/d/migrate-resolve <br />Migration name: {migration_name}<br />Database error code: {database_error_code}<br />Database error: {database_error}"<br /> |                                                                                                                                      |
```
Comment (Jolg42) [inline content/400-reference/200-api-reference/075-error-reference.mdx]:
```suggestion
| `P3016`   | "The fallback method for database resets failed, meaning Migrate could not clean up the database entirely. Original error: {error_code}\n{inner_error}"
|
| `P3017`   | "The migration {migration_name} could not be found. Please make sure that the migration exists, and that you included the whole name of the directory. (example: \"20201207184859_initial_migration\")"                                                                                                                           |
| `P3018`   | "A migration failed to apply. New migrations can not be applied before the error is recovered from. Read more about how to resolve migration issues in a production database: https://pris.ly/d/migrate-resolve <br />Migration name: {migration_name}<br />Database error code: {database_error_code}<br />Database error: {database_error}"                                                                                                                                    |
```

File: content/400-reference/200-api-reference/075-error-reference.mdx [hunk 4]
```diff
@@ -74,78 +140,15 @@ Prisma Client throws a `PrismaClientKnownRequestError` exception if the query en
...
+| `P3016`   | "The fallback method for database resets failed, meaning Migrate could not clean up the database entirely. Original error: {error_code}\n{inner_error}"
+|
+| `P3017`   | "The migration {migration_name} could not be found. Please make sure that the migration exists, and that you included the whole name of the directory. (example: \"20201207184859_initial_migration\")"                                                                                                                           |
+| `P3018`   | "A migration failed to apply. New migrations can not be applied before the error is recovered from. Read more about how to resolve migration issues in a production database: https://pris.ly/d/migrate-resolve <br />Migration name: {migration_name}<br />Database error code: {database_error_code}<br />Database error: {database_error}"                                                                                                                                    |
```
Comment (Jolg42) [inline content/400-reference/200-api-reference/075-error-reference.mdx]:
```suggestion
| `P3016`   | "The fallback method for database resets failed, meaning Migrate could not clean up the database entirely. Original error: {error_code}\n{inner_error}"<br />
|
| `P3017`   | "The migration {migration_name} could not be found. Please make sure that the migration exists, and that you included the whole name of the directory. (example: \"20201207184859_initial_migration\")"<br />
|
| `P3018`   | "A migration failed to apply. New migrations can not be applied before the error is recovered from. Read more about how to resolve migration issues in a production database: https://pris.ly/d/migrate-resolve <br />Migration name: {migration_name}<br />Database error code: {database_error_code}<br />Database error: {database_error}"<br />
|
```

File: content/400-reference/200-api-reference/075-error-reference.mdx [hunk 5]
```diff
@@ -74,78 +140,17 @@ Prisma Client throws a `PrismaClientKnownRequestError` exception if the query en
...
-| `P3017`   | "The migration {migration_name} could not be found. Please make sure that the migration exists, and that you included the whole name of the directory. (example: \"20201207184859_initial_migration\")"                                                                                                                                                                 |
+| `P3016`   | "The fallback method for database resets failed, meaning Migrate could not clean up the database entirely. Original error: {error_code}\n{inner_error}"<br />
+|
+| `P3017`   | "The migration {migration_name} could not be found. Please make sure that the migration exists, and that you included the whole name of the directory. (example: \"20201207184859_initial_migration\")"<br />
+|
+| `P3018`   | "A migration failed to apply. New migrations can not be applied before the error is recovered from. Read more about how to resolve migration issues in a production database: https://pris.ly/d/migrate-resolve <br />Migration name: {migration_name}<br />Database error code: {database_error_code}<br />Database error: {database_error}"<br />
+|
```
Comment (Jolg42) [inline content/400-reference/200-api-reference/075-error-reference.mdx]:
```suggestion
| `P3016`   | "The fallback method for database resets failed, meaning Migrate could not clean up the database entirely. Original error: {error_code}\n{inner_error}"<br /> |
| `P3017`   | "The migration {migration_name} could not be found. Please make sure that the migration exists, and that you included the whole name of the directory. (example: \"20201207184859_initial_migration\")"<br /> |
| `P3018`   | "A migration failed to apply. New migrations can not be applied before the error is recovered from. Read more about how to resolve migration issues in a production database: https://pris.ly/d/migrate-resolve <br />Migration name: {migration_name}<br />Database error code: {database_error_code}<br />Database error: {database_error}"<br /> |
```

**Comment by mhwelander** [comment]
Looks good, merging - thank you!

**Comment by Jolg42** [comment]
Yeah 💯

**Pull request merged**